java/src/jmri/jmrit/operations/trains/TrainEditBuildOptionsFrame.java

```diff
@@ -814,6 +814,34 @@ private boolean checkInput() {
                     .getMessage("CanNotSave"), JmriJOptionPane.ERROR_MESSAGE);
             return false;
         }
+        if ((change1Engine.isSelected() && !checkModel(modelEngine1Box, numEngines1Box, roadEngine1Box)) ||
+                (change2Engine.isSelected() && !checkModel(modelEngine2Box, numEngines2Box, roadEngine2Box))) {
+            return false;
+        }
+        return true;
+    }
+
+    private boolean checkModel(JComboBox<String> modelEngineBox, JComboBox<String> numEnginesBox,
+            JComboBox<String> roadEngineBox) {
+        String model = (String) modelEngineBox.getSelectedItem();
+        if (numEnginesBox.getSelectedItem().equals("0") || model.equals(NONE)) {
```

Comment:
No need to change, but a tip:

If you use "0".equals(something) instead of something.equals("0"), you will not get a NPE if something is null. For example if numEnginesBox doesn't have any item selected or if it's empty.

But if numEnginesBox always has items and it always has one of them selected, this is not an issue.

Reply:
Thanks for the tip